runner/server/handle.go

```diff
@@ -135,6 +136,41 @@ func (s *Server) handleApplyService() http.HandlerFunc {
 	}
 }
 
+func (s *Server) handleGetLogs() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		// Close the body of the request
+		defer utils.CloseTheCloser(r.Body)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		// get query params
+		vars := mux.Vars(r)
+		projectID := vars["project"]
+		taskID := vars["taskId"]
+		replicaID := vars["replicaId"]
+		follow, isFollowExists := r.URL.Query()["follow"]
+		if !isFollowExists {
+			follow[0] = "false"
+		}
+		isFollow, err := strconv.ParseBool(follow[0])
+		if err != nil {
+			logrus.Errorf("Failed to get service logs - %s", err.Error())
+			_ = utils.SendErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		utils.LogDebug("Get logs process started", "docker", "GetLogs", map[string]interface{}{"projectId": projectID, "taskId": taskID, "replicaId": replicaID, "isFollow": isFollow})
+
+		if err := s.driver.GetLogs(ctx, isFollow, projectID, taskID, replicaID, w, r); err != nil {
```

Comment:
I think `GetLogs` should return a reader. Internally you can use io.Pipe to achieve this.